src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -788,64 +931,28 @@ def main():
     gfu_service_ver = args.gfu_service_ver
     keep_job_dir = args.keep_job_dir
     catalog_admin = args.as_catalog_admin
-    load_id = args.load_id
-    if not load_id:
-        print("load_id is not provided. Generating a load_id ...")
-        load_id = uuid.uuid4().hex
+    load_id = args.load_id or uuid.uuid4().hex
     print(
         f"load_id is {load_id}.\n"
         f"Please keep using this load version until the load is complete!"
     )
-
     env = args.env
     kb_base_url = loader_common_names.KB_BASE_URL_MAP[env]
 
-    if workspace_id <= 0:
-        parser.error(f"workspace_id needs to be > 0")
-    if batch_size <= 0:
-        parser.error(f"batch_size needs to be > 0")
-    if cbs_max_tasks <= 0:
-        parser.error(f"cbs_max_tasks needs to be > 0")
+    _validate_arguments(workspace_id, batch_size, cbs_max_tasks)
 
-    uid = os.getuid()
     username = os.getlogin()
-
-    job_dir = loader_helper.make_job_dir(root_dir, username)
-    collection_source_dir = loader_helper.make_collection_source_dir(
-        root_dir, loader_common_names.DEFAULT_ENV, kbase_collection, source_version
-    )
-    collections_source_dir = loader_helper.make_collection_source_dir(
-        root_dir, env, kbase_collection, source_version
+    job_dir, collection_source_dir, collections_source_dir, source_dir = _prepare_directories(
```

Comment:
Would it be too difficult to rename collection_source_dir / collections_source_dir here? Those var names are way too close.

Reply:
👍

Reply:
Is it guaranteed the source dir is NCBI?

Reply:
source_dir is the WS source dir. (e.g. sourcedata/WS/<UPA>). Rename it as something like `ws_source_dir`?

Reply:
I'm confused now. If I understand the renaming:

collection_source_dir -> ncbi_coll_src_dir  (e.g. collectionssource/None/GTDB/...)
collections_source_dir -> ws_coll_src_dir (e.g. collectionssource/<env>/GTDB/...)

Is that right? If so ncbi_coll_src_dir doesn't seem appropriate

Reply:
that's correct. we use collectoin_source representing collectionssource dir. And source representing sourcedata dir. `ncbi_coll_src_dir` is confusing? what's your suggestion?

Reply:
yeah, ncbi doesn't appear in the directory path, and there's no guarantee in the future that the only external source will be ncbi, so at least to me the var name is confusing

Maybe `external coll_src_dir`?

Reply:
I see. 👍

Reply:
:+1: